Sources/Models/API/ProductAttributes.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  ProductAttributes.swift
+//  OpenFoodFacts
+//
+//  Created by Alexander Scott Beaty on 9/29/20.
+//
+import ObjectMapper
+
+/**
+ A model for Products usinig the [Product Attributes api](https://wiki.openfoodfacts.org/Product_Attributes)
+ */
+struct ProductAttributes {
+    var barcode: String?
+    var attributeGroups: [AttributeGroup]?
+
+    init?(map: Map) {
+        barcode <- map[ProductAttributesJson.BarcodeKey]
+        attributeGroups <- map[OFFJson.ProductKey + "." + ProductAttributesJson.AttributeGroupsKey]
+        // Fail init if some basic values are not present
+        if self.barcode == nil && self.attributeGroups == nil {
+            return nil
+        }
+    }
+}
```

Comment:
Now I am utilizing the `ProductAttributes` struct to hold the `attributeGroups` separately from the `product`. If you look at `ProductResponse.swift`, you'll see that `ProductAttributes` is now a property of `ProductResponse`.

As you can see in the screenshot below, the mapping is successful.
![ProductAttributes-top-level-struct-mapping-successful](https://user-images.githubusercontent.com/25991237/95665238-138a1500-0b14-11eb-9255-65556a6aad86.png)

Reply:
As you het the attributes together with the rest of the product data, I guess you need to integrate it into the product.

Reply:
How would you like me to do so?
 Right now, the `ProductAttributes` object has the barcode (`code`) and is itself a member of `ProductResponse` along with `Product`.
 So I was thinking, integrating it with the product, to me, simply means getting it from `ProductResponse` and passing it to `Product` before that is passed to the `onSuccess` callback near the end of `getProduct(byBarcode:)` in 'ProductService.swift' (line 152)
@aleene, suggestions?